llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -379,6 +381,26 @@ RISCVInstructionSelector::selectSHXADD_UWOp(MachineOperand &Root,
   return std::nullopt;
 }
 
+InstructionSelector::ComplexRendererFns
+RISCVInstructionSelector::selectVLOp(MachineOperand &Root) const {
+  MachineRegisterInfo &MRI =
+      Root.getParent()->getParent()->getParent()->getRegInfo();
+  assert(Root.isReg() && "Expected operand to be a Register");
+  MachineInstr *RootDef = MRI.getVRegDef(Root.getReg());
+
+  if (RootDef->getOpcode() == TargetOpcode::G_CONSTANT &&
+      RootDef->getOperand(1).getCImm()->getSExtValue() == RISCV::VLMaxSentinel)
+    // If the operand is a G_CONSTANT with value VLMaxSentinel, convert it
+    // to an immediate with value VLMaxSentinel. This is recognized specially by
+    // the vsetvli insertion pass.
+    return {
+        {[=](MachineInstrBuilder &MIB) { MIB.addImm(RISCV::VLMaxSentinel); }}};
+
+  // FIXME: Implement non-VLMAX case. ISEL will fail gracefully by returning
```

Comment:
What is preventing supporting this?

Reply:
I added support. I didn't add it because we weren't generating it, but it is good to have support going forward.